src/pybind/mgr/progress/module.py

```diff
@@ -194,15 +194,17 @@ def __init__(self, message, refs, add_to_ceph_s, start_epoch, active_clean_num):
         self._active_clean_num = active_clean_num
         self._refresh()
 
-    def global_event_update_progress(self, pg_dump):
+    def global_event_update_progress(self, pg_dump, log):
         # type: (Dict) -> None
         "Update progress of Global Recovery Event"
 
+        skipped_pgs = 0
         pgs = pg_dump['pg_stats']
         new_active_clean_num = 0
         for pg in pgs:
 
             if int(pg['reported_epoch']) < int(self._start_epoch):
+                skipped_pgs += 1
```

Comment:
was this caused by the test deleting pools, and those old stats sticking around too long?

If so, it'd be better to test for the pool still existing here, so we don't complete an event that hasn't had pg updates from non-active+clean pgs yet, but is still in progress. Better yet, we could delete removed pools from pg stats as soon as the pool is deleted - the OSD no longer sends updates after a pool is deleted

Reply:
looking at the mgr handling, it should be immediately removing pg stats when it handles the osdmap deleting the pool. perhaps there's a delay before the progress module notices because it's only updating the global progress event when there's a new pg_summary available, not when there's a new osdmap. rather than add more processing to a hot path, perhaps it's time to switch the progress module to update periodically instead of every time there's a new osd or pg map.